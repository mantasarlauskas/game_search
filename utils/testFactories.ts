import { Category } from 'utils/types';

export function getCategories(): Category[] {
    return [
        {
            id: 18893,
            name: 'Feral Interactive',
            slug: 'feral-interactive',
            games_count: 107,
            image_background: 'https://media.rawg.io/media/games/157/15742f2f67eacff546738e1ab5c19d20.jpg',
            games: [
                {
                    slug: 'rise-of-the-tomb-raider',
                    name: 'Rise of the Tomb Raider',
                },
                {
                    slug: 'shadow-of-mordor',
                    name: 'Middle-earth: Shadow of Mordor',
                },
                {
                    slug: 'hitman',
                    name: 'Hitman',
                },
                {
                    slug: 'bioshock-2',
                    name: 'BioShock 2',
                },
                {
                    slug: 'bioshock-remastered',
                    name: 'BioShock Remastered',
                },
                {
                    slug: 'xcom-enemy-unknown',
                    name: 'XCOM: Enemy Unknown',
                },
            ],
        },
        {
            id: 405,
            name: 'Ubisoft',
            slug: 'ubisoft',
            games_count: 297,
            image_background: 'https://media.rawg.io/media/games/5e4/5e4bff02098b2b6f824c68286d5da1a6.jpg',
            games: [
                {
                    slug: 'far-cry-3',
                    name: 'Far Cry 3',
                },
                {
                    slug: 'for-honor',
                    name: 'For Honor',
                },
                {
                    slug: 'watch-dogs',
                    name: 'Watch Dogs',
                },
                {
                    slug: 'far-cry-5',
                    name: 'Far Cry 5',
                },
                {
                    slug: 'assassins-creed-origins',
                    name: 'Assassin\'s Creed Origins',
                },
                {
                    slug: 'rayman-legends',
                    name: 'Rayman Legends',
                },
            ],
        },
    ];
}
