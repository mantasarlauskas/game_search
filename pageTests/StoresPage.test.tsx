import StoresPage from 'pages/stores';
import { render, screen } from '@testing-library/react';
import { getCategories } from 'testing/testFactories';

jest.mock('utils/fetch');

const mockedCategories = getCategories();
jest.mock('pages/_app', () => ({
    useAppContext() {
        return {
            stores: mockedCategories,
        };
    },
}));

describe('<StoresPage />', () => {
    it('matches snapshot', () => {
        const { container } = render(<StoresPage />);
        expect(container).toMatchSnapshot();
    });

    it('renders content', () => {
        render(<StoresPage />);
        expect(screen.getByText('Stores')).toBeInTheDocument();
        expect(screen.getByText('Feral Interactive')).toBeInTheDocument();
        expect(screen.getByText('Ubisoft')).toBeInTheDocument();
    });
});
