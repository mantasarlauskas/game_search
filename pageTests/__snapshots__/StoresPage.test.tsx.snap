// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoresPage /> matches snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <h1
      class="title"
    >
      Stores
    </h1>
    <div
      class="categories"
    >
      <div
        class="root"
      >
        <div>
          <div
            class="name"
          >
            Feral Interactive
          </div>
        </div>
        <div
          class="heading"
        >
          <div
            class="title"
          >
            Games
          </div>
          <div
            class="count"
          >
            107
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            Rise of the Tomb Raider
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            Middle-earth: Shadow of Mordor
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            Hitman
          </div>
        </div>
      </div>
      <div
        class="root"
      >
        <div>
          <div
            class="name"
          >
            Ubisoft
          </div>
        </div>
        <div
          class="heading"
        >
          <div
            class="title"
          >
            Games
          </div>
          <div
            class="count"
          >
            297
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            Far Cry 3
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            For Honor
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            Watch Dogs
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
